src/SurveyService.Business/Commands/Group/GetGroupCommand.cs

```diff
@@ -0,0 +1,130 @@
+﻿using LT.DigitalOffice.Kernel.Extensions;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.Models.Broker.Models;
+using LT.DigitalOffice.SurveyService.Broker.Requests.Interfaces;
+using LT.DigitalOffice.SurveyService.Business.Commands.Group.Interfaces;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.SurveyService.Mappers.Responses.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Db;
+using LT.DigitalOffice.SurveyService.Models.Dto.Models;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Group.Filters;
+using LT.DigitalOffice.SurveyService.Models.Dto.Responses.Group;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+
+namespace LT.DigitalOffice.SurveyService.Business.Commands.Group;
+
+public class GetGroupCommand : IGetGroupCommand
+{
+  private readonly IGroupRepository _repository;
+  private readonly IGroupResponseMapper _groupResponseMapper;
+  private readonly IResponseCreator _responseCreator;
+  private readonly IUserService _userService;
+  private readonly IQuestionInfoMapper _questionInfoMapper;
+  private readonly IOptionInfoMapper _optionInfoMapper;
+  private readonly IUserAnswerInfoMapper _userAnswerInfoMapper;
+  private readonly IHttpContextAccessor _httpContextAccessor;
+
+  public GetGroupCommand(
+    IGroupRepository groupRepository,
+    IGroupResponseMapper groupResponseMapper,
+    IResponseCreator responseCreator,
+    IUserService userService,
+    IQuestionInfoMapper questionInfoMapper,
+    IOptionInfoMapper optionInfoMapper,
+    IUserAnswerInfoMapper userAnswerInfoMapper,
+    IHttpContextAccessor httpContextAccessor)
+  {
+    _repository = groupRepository;
+    _groupResponseMapper = groupResponseMapper;

```

Comment:
use in command only group mapper;
question mapper should be inside group mapper;
option mapper - inside question mapper
and so on

Reply:
you receive from repo dbGroup, then check IsPrivate, IsAnonymous, Deadline... if result is not valid, return Forbidden and don't need mapping
if result is valid, you map dbGroup to GroupResponse

follow comments with *** mark (but check it - may be i miss something)

Reply:
Done